cmd/sonobuoy/app/results.go

```diff
@@ -254,6 +317,86 @@ func printResultsDetails(treePath []string, o *results.Item, input resultsInput)
 }
 
+// filterAndSortHealthInfoDetails takes a copy of a slice of HealthInfoDetails, 
+// discards the ones that are healthy,
+// then sorts the remaining entries, 
+// and finally sorts them by namespace and name
+func filterAndSortHealthInfoDetails(details []discovery.HealthInfoDetails) ([]discovery.HealthInfoDetails) {
+	result := make([]discovery.HealthInfoDetails, len(details))
+	var idx int
+	for _, detail := range details {
+		if !detail.Healthy {
+			result[idx] = detail 
+			idx++
+		}
+	}
+	result = result[:idx]
+	isLess := func (i,j int) bool {
+		if result[i].Namespace == result[j].Namespace {
+			return result[i].Name < result[j].Name
+		} else {
+			return result[i].Namespace < result[j].Namespace
+		}
+	}
+	sort.Slice(result, isLess)
+	return result
+}
+
+
+// printClusterHealthResultsSummary prints the summary of the "fake" plugin for health summary,
+// tryingf to emulate the format of printResultsSummary
+func printClusterHealthResultsSummary(summary discovery.ClusterSummary) error {
+	fmt.Println("Run Details:")
+	fmt.Printf("API Server version: %s\n", summary.APIVersion)
+
+	fmt.Printf("Node health: %d/%d", summary.NodeHealth.Healthy, summary.NodeHealth.Total)
+	//Print the percentage only if Total is not 0 to avoid division by zero errors
+	if summary.NodeHealth.Total != 0 {
+		fmt.Printf(" (%d%%)", 100 * summary.NodeHealth.Healthy / summary.NodeHealth.Total)
+	}
+	fmt.Println()
+	//Details of the failed pods. Checking the slice length to avoid trusting the Total
+	if len(summary.NodeHealth.Details) > 0 && summary.NodeHealth.Healthy < summary.NodeHealth.Total {
+		fmt.Println("Details for failed nodes:")
+		nodes := filterAndSortHealthInfoDetails(summary.NodeHealth.Details)
+		for _, node := range nodes {
+			fmt.Printf("%s Ready:%s: %s: %s\n", node.Name, node.Ready, node.Reason, node.Message)
+		}
+		fmt.Println()
+	}
+
+	//It might be nice to group pods by namespace.
+	//Also here, use len instead of trusting Total
+	if len(summary.PodHealth.Details) > 0 {
+		fmt.Printf("Pods health: %d/%d", summary.PodHealth.Healthy, summary.PodHealth.Total)
+		//Print the percentage only if Total is not 0 to avoid division by zero errors
+		if summary.PodHealth.Total != 0 {
+			fmt.Printf(" (%d%%)", 100 * summary.PodHealth.Healthy / summary.PodHealth.Total)
+		}
+		fmt.Println()
+		if summary.PodHealth.Healthy < summary.PodHealth.Total {
+			fmt.Println("Details for failed pods:")
+			pods := filterAndSortHealthInfoDetails(summary.PodHealth.Details)
+			//And then print them, sorted by namespace
+			for _, pod := range pods {
+				fmt.Printf("%s/%s Ready:%s: %s: %s\n", pod.Namespace, pod.Name, pod.Ready, pod.Reason, pod.Message)
+			}
+		}
+	}
+
+	if len(summary.ErrorInfo) > 0 {
+		fmt.Println("Errors detected in files:")
+		for errorType := range summary.ErrorInfo {
+			fmt.Printf("%s:\n", errorType)
+			for fileName := range summary.ErrorInfo[errorType] {
+				fmt.Printf("%s:%d\n", fileName, summary.ErrorInfo[errorType][fileName])
```

Comment:
I didnt realize it until i was trying to verify the output, but we shouldnt use `filename:errorCount`; it looks exactly like the format used for `filename:line`. I was thinking there was a bug identifying errors but it was giving me an error count and not the line number of each detected error.

I think it would be most clear to say `filename (%d found)`

Reply:
I think we can also put an issue to touch up the functionality slightly so we trace out what patterns did match/where. I'm looking at some of the output and having trouble identifying the warnings its saying it found. That doesn't have to be this PR though.

Reply:
Indeed, the first time I saw the description in the issue itself I also though these were supposed to be line numbers.
I've changed it to look like the output of `grep -l <regex> | sort | uniq -c | sort -n`, see if it looks good, otherwise I'll change it to what you suggested.
It would look something like this:
```
Errors:
22 podlogs/kube-system/kube-controller-manager-kind-control-plane/logs/kube-controller-manager.txt
22 podlogs/kube-system/kube-scheduler-kind-control-plane/logs/kube-scheduler.txt
 2 podlogs/kube-system/kube-apiserver-kind-control-plane/logs/kube-apiserver.txt
```